operations/rates.md

```diff
@@ -49,222 +52,159 @@ Note this operation uses [Pagination](../guidelines/pagination.md) and supports
-        }
-    ]
-}
-```
-
-| Property | Type | Contract | Description |
-| :-- | :-- | :-- | :-- |
-| `Rates` | array of [Rate](#rate) | required | Rates of the default service. |
-| ~~`RateGroups`~~ | ~~array of [Rate group](#rate-group)~~ | ~~required~~ | ~~Rate groups of the default service.~~ **Deprecated!** |
-| `AvailabilityBlockAssignments` | array of [Availability block assignment](#availability-block-assignment) | optional | Shows which rates relate to which availability blocks. |
-| `Cursor` | string | optional | Unique identifier of the item one newer in time order than the items to be returned. If Cursor is not specified, i.e. null, then the latest or most recent items will be returned. |
-
-## Add rates
-
-Adds rates to the enterprise. Note this operation supports [Portfolio Access Tokens](../guidelines/multi-property.md). Rate type of `AvailabilityBlock` cannot be created via this operation.
-
-### Request
-
-`[PlatformAddress]/api/connector/v1/rates/add`
-
-```javascript
-{
-    "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
-    "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
-    "Client": "Sample Client 1.0.0",
-    "EnterpriseId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
-    "Rates":[
-      {
-          "ServiceId": "bd26d8db-86da-4f96-9efc-e5a4654a4a94",
-          "RateGroupId": "b9f25a45-9b9a-4b33-99bd-b06f008eb6f5",
-          "IsEnabled": true,
-          "Type": "Public",
-          "AccountingCategoryId": "3620c660-a4ec-4e0f-a0bc-b06f008eb8bf",
-          "Names": { "EN": "My rate" },
-          "ExternalIdentifier": "D001",
-          "PricingType": "DependentRatePricing",          
-          "Pricing": {
-              "DependentRatePricing": {
-                  "BaseRateId": "1a1ddd3b-e106-4a70-aef1-54a75b483943",
-                  "RelativeAdjustment": 0.15,
-                  "AbsoluteAdjustment": 10.0
-              }
-          }
-      }
+  "Rates": [
+    {
+      "BaseRateId": null,
+      "BusinessSegmentId": null,
+      "GroupId": "c8b866b3-be2e-4a47-9486-034318e9f393",
+      "Id": "ed4b660b-19d0-434b-9360-a4de2ea42eda",
+      "ServiceId": "bd26d8db-86da-4f96-9efc-e5a4654a4a94",
+      "IsActive": true,
+      "IsEnabled": true,
+      "IsPublic": true,
+      "Name": "Fully Flexible",
+      "ShortName": "FF",
+      "ExternalNames": {
+        "en-US": "Long Stay Flexible Rate"
+      },
+      "ExternalIdentifier": "D001"
+    }
+  ],
+  "RateGroups": [
+    {
+      "Id": "c8b866b3-be2e-4a47-9486-034318e9f393",
+      "ServiceId": "bd26d8db-86da-4f96-9efc-e5a4654a4a94",
+      "IsActive": true,
+      "Name": "Default",
+      "ExternalIdentifier": "RG001"
+    }
+  ],
+  "AvailabilityBlockAssignments": [
+    {
+      "RateId": "ed4b660b-19d0-434b-9360-a4de2ea42eda",
+      "AvailabilityBlockId": "12806ae8-9c15-44c7-9a44-afae01289928"
+    }
   ]
 }
 ```
 
 | Property | Type | Contract | Description |
```

Comment:
What happened to `AvailabilityBlockAssignments` in the response?

Reply:
@MikeAdamsMews its marked as `Internal implementation` and we dont render these properties to documentation. See [HERE] (https://github.com/MewsSystems/mews/blob/1e5667ed8a1d714a4721ddcaa209823d1b4967c0/src/Modules/ConnectorApi/Mews.ConnectorApi.Contracts/V1/Dto/Services/Pricing/RateParameters.cs#L53) Is it ok or do you think it shouldnt be marked as InternalImplementation?

Reply:
The problem is that the field is shown in the current published documentation, I assume that is a mistake. If that is true and it's currently set to`Internal implementation` and no user can actually use it in the API, then I think okay we can just remove it from the docs as we are simply correcting a mistake. However, if it is usable by clients then it counts as a deprecation instead.

Reply:
@jnv any thoughts? It's tough to judge the correct solution. I give more weight to the code, so I don't think it should be in the documentation.

Reply:
AvailabilityBlockAssignments were explicitly documented in https://github.com/MewsSystems/gitbook-connector-api/commit/419ceb701dc08dcfb38261099bfed5bcc3e367ad so it should be here. We just forgot to remove InternalImplementation.

Reply:
ok, fixed [here](https://github.com/MewsSystems/gitbook-connector-api/pull/635/commits/6cccccdfb5ec7d2085e1803d729d2a4bbd1b9c88). Removed `internalImplementation `and removed depricated from `extent `since it doesnt make sence anymore